### PR TITLE
Serialize logging

### LINKED ISSUE #268: LoggingPolicy messages are interleaved
Logging messages are interleaved when there are multiple threads. Sometimes the messages even get interrupted in the middle of logging a log line.

We probably need to run the logger on its own thread?

### PR DESCRIPTION
**Pull request by bsiegel**
Fixes #268

### CODE REVIEW

File: sdk/core/AzureCore/Source/Pipeline/Policies/LoggingPolicy.swift [hunk 1]
```diff
@@ -72,9 +76,24 @@ public class LoggingPolicy: PipelineStage {
     // MARK: PipelineStage Methods
 
     public func on(request: PipelineRequest, then completion: @escaping OnRequestCompletionHandler) {
-        var returnRequest = request.copy()
```
Comment (bsiegel) [inline sdk/core/AzureCore/Source/Pipeline/Policies/LoggingPolicy.swift:75]: In testing with the demo app, it didn't seem like this was necessary - just updating the context on the provided `request` worked, the response logger was able to get the request start time back out of the context just fine.

File: sdk/core/AzureCore/Source/ClientLogger.swift [hunk 2]
```diff
@@ -156,15 +156,15 @@ public class OSLogger: ClientLogger {
...
         self.osLogger = osLogger
     }
 
     public init(
-        subsystem: String = "com.azure",
-        category: String,
-        level: ClientLogLevel = .info
+        forSubsystem subsystem: String = "com.azure",
+        withCategory category: String,
+        atLevel level: ClientLogLevel = .info
```
Comment (tjprescott) [inline sdk/core/AzureCore/Source/ClientLogger.swift:167]: Any specific reason for this change?

Comment (bsiegel) [reply sdk/core/AzureCore/Source/ClientLogger.swift:167]: No, just that this was the first swift code I wrote, and I probably didn't understand the whole internal vs external name thing. I think this is what I would have called it if I'd had the benefit of hindsight that I do now.

Comment (tjprescott) [reply sdk/core/AzureCore/Source/ClientLogger.swift:167]: It just seems that the old style was more consistent with the internal implementation classes:
```Swift
OSLog(subsystem: subsystem, category: category)
NSLogger(tag: "MyAppClient", level: .debug)
```

Comment (bsiegel) [reply sdk/core/AzureCore/Source/ClientLogger.swift:167]: I don't have a strong opinion, I guess the underlying logger APIs don't really feel particularly "swifty" either. But maybe you're right, better to be consistent with the existing Foundation classes even if they feel less idiomatic. I'll remove that change.

**Pull request merged**